### PR TITLE
[backend] Update API routes

### LINKED ISSUE #20: [Discord-Bot] Integrate Discord Bot with Backend
- [x] Integrate /register/ route to receive userHash and return it to user.
- [ ] Integrate WebSockets to show the status of the user on discord.

### LINKED ISSUE #21: [Deployment] Deploy backend
Deploy backend on Heroku and change the baseUrl everywhere to the generated REST API endpoint.

### PR DESCRIPTION
**Pull request by rish07**
Closes #21 and #20 

Base URL for the backend: https://kaal-backend.herokuapp.com/

### CODE REVIEW

**Comment by rish07** [comment]
@pandafy can we merge this into master and update the bot on Heroku?

File: moropy_backend/app.py [hunk 1]
```diff
@@ -12,17 +12,24 @@ def welcome():
 @app.route('/register/', methods=['POST'])
 def register_user():
     input_json = request.get_json(force=True)
-    print("USER ID", input_json['user_id'])
+
+    print("USER ID", input_json['userId'])
     print("ROLES", input_json['roles'])
```
Comment (pandafy) [inline moropy_backend/app.py]: Remove these print messages.

File: moropy_backend/app.py [hunk 2]
```diff
@@ -12,17 +12,24 @@ def welcome():
...
-@app.route('/<string:userHash>', methods=['GET'])
-def get_user_details(userHash):
-    print(userHash)
-    resp = get_user(userHash)
-    return jsonify({'user': resp})
+@app.route('/validate/', methods=['POST'])
+def get_user_details():
+    input_json = request.get_json(force=True)
+
+    print(input_json['userHash'])
```
Comment (pandafy) [inline moropy_backend/app.py]: Remove `print` statements.

File: moropy_backend/app.py [hunk 3]
```diff
@@ -40,7 +47,9 @@ def activity():
             return jsonify({"msg": "Error"})
 
 
-@app.route('/status/', methods=['POST'])
+@app.route('/status/', methods=['GET', 'POST'])
+
+
```
Comment (pandafy) [inline moropy_backend/app.py]: Remove these unnecessary blank lines

File: moropy_backend/firebase_services.py [hunk 4]
```diff
@@ -12,23 +12,28 @@
 batch = db.batch()
 
 
-def upload(userID, roles):
+def upload(userID, roles, userName):
     print('working')
```
Comment (pandafy) [inline moropy_backend/firebase_services.py]: Remove `print` statements.

File: moropy_backend/firebase_services.py [hunk 5]
```diff
@@ -12,23 +12,28 @@
...
-    doc_ref = user_ref.document(f'{userHash}')
-    doc_ref.set({u'roles': roles, u'discordId': userID})
+    doc_ref = user_ref.document(f"{userHash}")
+    doc_ref.set({u'roles': roles, u'discordId': userID, u'userName': userName})
+
     return userHash
 
 
 def get_user(userHash):
     print("getting user")
```
Comment (pandafy) [inline moropy_backend/firebase_services.py]: Remove `print` statements.

File: moropy_backend/firebase_services.py [hunk 6]
```diff
@@ -12,23 +12,28 @@
...
+
+    print(userHash)
```
Comment (pandafy) [inline moropy_backend/firebase_services.py]: Remove `print` statements.

File: moropy_backend/firebase_services.py [hunk 7]
```diff
@@ -12,23 +12,28 @@
...
     users = user_ref.stream()
     for user in users:
         if user.id == userHash:
+
```
Comment (pandafy) [inline moropy_backend/firebase_services.py]: Remove blank line and print statement

File: moropy_backend/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,37 @@
+CacheControl==0.12.6
```
Comment (pandafy) [inline moropy_backend/requirements.txt]: Instead of dumping all dependencies, we should just keep direct dependencies.

File: moropy_bot/bot.py [hunk 9]
```diff
@@ -2,12 +2,15 @@
...
 
 import discord
+import requests
 from discord.ext import commands
 from dotenv import load_dotenv
 
 load_dotenv()
 TOKEN = os.getenv('DISCORD_TOKEN')
 
+baseURL = "https://kaal-backend.herokuapp.com/"
```
Comment (pandafy) [inline moropy_bot/bot.py]: Let's use an environment variable for this.

File: moropy_bot/bot.py [hunk 10]
```diff
@@ -24,8 +27,22 @@ async def on_message(ctx):
...
+    )
+
     await channel.send(response)
+    r = requests.post(
+        url=baseURL + '/register/',
+        json={"userId": str(user.id), "userName": str(user), "roles": roles},
+    )
+
+    res = r.json()
+    print(res)
```
Comment (pandafy) [inline moropy_bot/bot.py]: Remove `print` statements.

**Review by pandafy — CHANGES_REQUESTED** [review]
@rish07 @zerefwayne let's keep the code tidy and remove any print statements before committing to GitHub

**Comment by pandafy** [comment]
> @pandafy can we merge this into master and update the bot on Heroku?

I will make some changes here and will update the bot on Heroku.

**Comment by rish07** [comment]
> > @pandafy can we merge this into master and update the bot on Heroku?
> 
> I will make some changes here and will update the bot on Heroku.

Are you making all these changes you requested? @pandafy

**Comment by pandafy** [comment]
> > > @pandafy can we merge this into master and update the bot on Heroku?
> > 
> > 
> > I will make some changes here and will update the bot on Heroku.
> 
> Are you making all these changes you requested? @pandafy

Yes

**Review by rish07 — COMMENTED** [review]
Fixed the issues @pandafy

**Review by pandafy — APPROVED** [review]
LGTM :+1: @rish07 please resolve merge conflicts and merge at your discretion.

**Pull request merged**